suzieq/engines/pandas/path.py

```diff
@@ -400,6 +428,51 @@ def _get_underlay_nexthop(self, hostname: str, vtep_list: list,
 
         return result
 
+    def _handle_recursive_route(self, df: pd.DataFrame,
+                                max_recurse: int) -> pd.DataFrame:
+        '''Handle recursive routes which are routes without an OIF.
+
+        NXOS, Panos, IOS and others have this scenario when showing BGP routes
+
+        Args:
+            df (pd.DataFrame): The route DF with the routes to resolve
+            max_recurse(int): Recurse this many times to resolve a route
+
+        Returns:
+            pd.DataFrame: The original dataframe with the OIF filled in
+        '''
+        for row in df.itertuples():
+            oifs = []
+            nhops = []
+            for nhop in row.nexthopIps:
+                rrdf = self._get_table_sqobj('routes') \
+                           .lpm(namespace=[row.namespace],
+                                hostname=[row.hostname], address=[nhop],
+                                vrf=[row.vrf])
+
+                if not rrdf.empty:
+                    if rrdf.oifs.tolist() == []:
+                        # Recurse once more
+                        max_recurse -= 1
+                        if max_recurse:
+                            rrdf = self._handle_recursive_route(rrdf,
+                                                                max_recurse)
```

Comment:
Suppose that we have `row.nexthopIps` with 3 elements. Everyone of them has a route table such as `rrdf` is not empty. Suppose also that `max_recurse = 2`.
Doing `max_recurse -= 1` will never trigger the recursion for the 3rd element of `row.nexthopIps` and I don't think it is what you expected. I suggest to remove the `max_recurse -= 1` and write `rrdf = self._handle_recursive_route(rrdf, max_recurse-1)`